cocos/core/event/async-delegate.ts

```diff
@@ -104,6 +104,11 @@ export class AsyncDelegate<T extends (...args: any) => (Promise<void> | void) =
      * @returns @en The promise awaiting all async callback resolved. @zh 等待所有异步回调结束的 Promise 对象。
      */
     public dispatch (...args: Parameters<T>) {
-        return Promise.all(this._delegates.map((func) => func(...arguments)).filter(Boolean));
+        if (window.oh) {
+            // TODO(qgh):OpenHarmony currently does not support arguments
+            return Promise.all(this._delegates.map((func) => func(args)).filter(Boolean));
```

Comment:
```suggestion
            return Promise.all(this._delegates.map((func) => func(...args)).filter(Boolean));
```

Reply:
Strange, I think we need the spread syntax here
https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
Strange, I think we need the spread syntax here

Reply:
Strange, I think we need the spread syntax here
https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Operators/Spread_syntax

Reply:
This is a temporary fix that will be restored later.